src/daemonizer/posix_fork.cpp

```diff
@@ -21,15 +21,34 @@
 namespace posix {
 
 namespace {
-  void quit(std::string const & message)
+  void quit(const std::string & message)
   {
     LOG_ERROR(message);
     throw std::runtime_error(message);
   }
 }
 
-void fork()
+void fork(const std::string & pidfile)
 {
+  // If a PID file is specified, we open the file here, because
+  // we can't report errors after the fork operation.
+  // When we fork, we close thise file in each of the parent
+  // processes.
+  // Only in the final child process do we write the PID to the
+  // file (and close it).
+  std::ofstream pidfs;
+  if (! pidfile.empty ())
+  {
+    if (access(pidfile.c_str(), F_OK) == 0)
```

Comment:
Got it. OK so this introduces another problem in that you are only allowing one daemon to run. I don't feel you should enforce this but rather print an error message and let the process continue without writing to the pid file.